pkg/log/log.go

```diff
@@ -150,6 +151,29 @@ func format(msg, level string, ts time.Time, appInfo Marshaler, mm Many) string
 		generateFatalFields(entry)
 	}
 
+	// Attempt to map the caller of the log function into the "source" field for identifying if a service or a module
+	// that the service is using is sending logs (costing money).
+	caller, err := callerinfo.GetCallerFunction(2)
+	switch {
+	case err != nil:
+		entry["source"] = "error"
+	case strings.HasPrefix(caller, "github.com/"):
+		// Example response: github.com/getoutreach/gobox/pkg/callerinfo.Test_Callers
+		splits := strings.Split(caller, "/")
+		// We're going for extracting "getoutreach/gobox" in the above example
+		if len(splits) >= 3 {
+			entry["source"] = splits[1] + "/" + splits[2]
+		} else {
+			entry["source"] = splits[1]
+		}
+	case strings.HasPrefix(caller, "main"):
+		// Main entry point for the app shows up as "main.[funcname]", so just use the app name since we unfortunately
+		// don't have a full package name
+		entry["source"] = app.Info().Name
```

Comment:
Prefix with `app:` to help disambiguate?

Reply:
i like that, will do

Reply:
Or should `source` be the module name as it appears to Go? That goes back to my earlier comment about https://pkg.go.dev/runtime/debug#ReadBuildInfo.